R/tbl_shift.R

```diff
@@ -0,0 +1,307 @@
+#'   `"header"`: stratum labels are placed in a header row above the tabulations.
+#' @param strata_label (`string`)\cr
+#'   A glue-string that inserts stratum level. Default is `'{strata}'`, and
+#'   `{n}` is also available to insert.
+#' @param header (`string`)\cr
+#'   String that is passed to `gtsummary::modify_header(all_stat_cols() ~ header)`.
+#' @param label ([`formula-list-selector`][gtsummary::syntax])\cr
+#'   Used to specify the labels for the `strata` and `variable` columns.
+#'   Default is to use the column label attribute.
+#' @param nonmissing,nonmissing_text,... Argument passed to `tbl_roche_summary()`.
+#'   See details below for call details to `tbl_roche_summary()`.
+#' @param x (`tbl_shift`)\cr
+#'   Object of class `'tbl_shift'`.
+#' @param col_label (`string`)\cr
+#'   String indicating the column label. Default is `"All Participants \nN = {gtsummary::style_number(n)}"`
+#'
+#' @returns a 'gtsummary' table
+#' @name tbl_shift
+#'
+#' @details
+#' Broadly, this function is a wrapper for chunk below with some additional
+#' calls to `gtsummary::modify_*()` function to update the table's
+#' headers, indentation, column alignment, etc.
+#'
+#' ```r
+#' gtsummary::tbl_strata2(
+#'   data = data,
+#'   strata = strata,
+#'    ~ tbl_roche_summary(.x, include = variable, by = by)
+#' )
+#' ```
+#'
+#' @examples
+#' library(dplyr, warn.conflicts = FALSE)
+#'
+#' # subsetting ADLB on one PARAM, and the highest grade
+#' adlb <- pharmaverseadam::adlb |>
+#'   select("USUBJID", "TRT01A", "PARAM", "PARAMCD", "ATOXGRH", "BTOXGRH", "VISITNUM") |>
+#'   mutate(TRT01A = factor(TRT01A)) |>
+#'   filter(PARAMCD %in% c("CHOLES", "GLUC")) |>
+#'   slice_max(by = c(USUBJID, PARAMCD), order_by = ATOXGRH, n = 1L, with_ties = FALSE) |>
+#'   labelled::set_variable_labels(
+#'     BTOXGRH = "Baseline  \nNCI-CTCAE Grade",
+#'     ATOXGRH = "Post-baseline  \nNCI-CTCAE Grade"
+#'   )
+#' adsl <- pharmaverseadam::adsl[c("USUBJID", "TRT01A")] |>
+#'   filter(TRT01A != "Screen Failure")
+#'
+#' # Example 1 ----------------------------------
+#' # tabulate baseline grade by worst grade
+#' tbl_shift(
+#'   data = filter(adlb, PARAMCD %in% "CHOLES"),
+#'   strata = BTOXGRH,
+#'   variable = ATOXGRH,
+#'   by = TRT01A,
+#'   data_header = adsl
+#' )
+#'
+#' # Example 2 ----------------------------------
+#' # same as Ex1, but with the stratifying variable levels in header rows
+#' adlb |>
+#'   filter(PARAMCD %in% "CHOLES") |>
+#'   labelled::set_variable_labels(
+#'     BTOXGRH = "Baseline NCI-CTCAE Grade",
+#'     ATOXGRH = "Post-baseline NCI-CTCAE Grade"
+#'   ) |>
+#'   tbl_shift(
+#'     data = ,
+#'     strata = BTOXGRH,
+#'     variable = ATOXGRH,
+#'     strata_location = "header",
+#'     by = TRT01A,
+#'     data_header = adsl
+#'   )
+#'
+#' # Example 3 ----------------------------------
+#' # same as Ex2, but with two labs
+#' adlb |>
+#'   filter(PARAMCD %in% "CHOLES") |>
```

Comment:
```suggestion
```

I'm wondering if the filtering should be removed in this step to demo this example - the dataset would only have two labs anyways so the nesting by lab may be clearer.

Reply:
oh yes! thank you!!